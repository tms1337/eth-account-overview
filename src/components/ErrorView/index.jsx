import React from "react";
import { colors } from "../../config/ui";

const ErrorView = ({error}) => {
  return <div style={styles.root}>{error}</div>;
}

const styles = {
  root: {
    color: colors.ERROR
  }
}

export default ErrorView;