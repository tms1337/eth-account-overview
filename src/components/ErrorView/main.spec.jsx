import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import ErrorView from ".";

describe("ErrorView", () => {
  it("renders snapshot", () => {
    const component = renderer.create(<ErrorView />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays correct error value", () => {
    const error = "An error has occured";
    const component = mount(<ErrorView error={error} />);

    expect(component.contains(error)).toBe(true);
  });

  it("displays correct different error value", () => {
    const error = "A different error has occured";
    const component = mount(<ErrorView error={error} />);

    expect(component.contains(error)).toBe(true);
  });
});
