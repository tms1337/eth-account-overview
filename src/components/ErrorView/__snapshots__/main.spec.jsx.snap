// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders snapshot 1`] = `
<div
  style={
    Object {
      "color": "#ff5b5b",
      "wordBreak": "break-all",
    }
  }
/>
`;
